swatch-producer-azure/src/main/java/com/redhat/swatch/azure/service/AzureMarketplaceClientFactory.java

```diff
@@ -58,13 +58,21 @@ public List<AzureClient> createClientForEachTenant() {
         .map(
             client -> {
               try {
-                var api =
-                    QuarkusRestClientBuilder.newBuilder()
-                        .baseUri(new URI(azureMarketplaceProperties.getMarketplaceBaseUrl()))
-                        .register(
-                            new AzureMarketplaceHeaderProvider(
-                                azureMarketplaceProperties, client, oidcClients))
-                        .build(AzureMarketplaceApi.class);
+                AzureMarketplaceApi api;
+                if (azureMarketplaceProperties.isDisableAzureOidc()) {
+                  api =
+                      QuarkusRestClientBuilder.newBuilder()
+                          .baseUri(new URI(azureMarketplaceProperties.getMarketplaceBaseUrl()))
+                          .build(AzureMarketplaceApi.class);
+                } else {
+                  api =
+                      QuarkusRestClientBuilder.newBuilder()
+                          .baseUri(new URI(azureMarketplaceProperties.getMarketplaceBaseUrl()))
+                          .register(
+                              new AzureMarketplaceHeaderProvider(
+                                  azureMarketplaceProperties, client, oidcClients))
+                          .build(AzureMarketplaceApi.class);
+                }
                 return new AzureClient(client.getClientId(), api);
```

Comment:
```suggestion
                var apiBuilder = QuarkusRestClientBuilder.newBuilder()
                          .baseUri(new URI(azureMarketplaceProperties.getMarketplaceBaseUrl()));
                if (!azureMarketplaceProperties.isDisableAzureOidc()) {
                  apiBuilder.register(
                              new AzureMarketplaceHeaderProvider(
                                  azureMarketplaceProperties, client, oidcClients));
                }
                return new AzureClient(client.getClientId(), apiBuilder.build(AzureMarketplaceApi.class));
```

I think it's clearer to use the builder in a common place instead.

Reply:
Makes sense, thanks! updated